## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 460388`
Commit: [`24a36b141b5099f6900bede6402cb5e6a66d8e34`](https://github.com/NixOS/nixpkgs/commit/24a36b141b5099f6900bede6402cb5e6a66d8e34) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/24a36b141b5099f6900bede6402cb5e6a66d8e34..pull/460388/head))
Merge: [`bc721fe36d20024015d7b70916e6e707ae563558`](https://github.com/NixOS/nixpkgs/commit/bc721fe36d20024015d7b70916e6e707ae563558)

Logs: https://github.com/Aleksanaa/nixpkgs-review-gha/actions/runs/19432328699


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>python312Packages.aioice</li>
    <li>python312Packages.aioice.dist</li>
    <li>python312Packages.aiortc</li>
    <li>python312Packages.aiortc.dist</li>
    <li>python312Packages.pylibsrtp</li>
    <li>python312Packages.pylibsrtp.dist</li>
    <li>python313Packages.aioice</li>
    <li>python313Packages.aioice.dist</li>
    <li>python313Packages.aiortc</li>
    <li>python313Packages.aiortc.dist</li>
    <li>python313Packages.pylibsrtp</li>
    <li>python313Packages.pylibsrtp.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>python312Packages.aioice</li>
    <li>python312Packages.aioice.dist</li>
    <li>python312Packages.aiortc</li>
    <li>python312Packages.aiortc.dist</li>
    <li>python312Packages.pylibsrtp</li>
    <li>python312Packages.pylibsrtp.dist</li>
    <li>python313Packages.aioice</li>
    <li>python313Packages.aioice.dist</li>
    <li>python313Packages.aiortc</li>
    <li>python313Packages.aiortc.dist</li>
    <li>python313Packages.pylibsrtp</li>
    <li>python313Packages.pylibsrtp.dist</li>
  </ul>
</details>

---
### `x86_64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python312Packages.aiortc</li>
    <li>python312Packages.aiortc.dist</li>
    <li>python313Packages.aiortc</li>
    <li>python313Packages.aiortc.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>python312Packages.aioice</li>
    <li>python312Packages.aioice.dist</li>
    <li>python312Packages.pylibsrtp</li>
    <li>python312Packages.pylibsrtp.dist</li>
    <li>python313Packages.aioice</li>
    <li>python313Packages.aioice.dist</li>
    <li>python313Packages.pylibsrtp</li>
    <li>python313Packages.pylibsrtp.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>python312Packages.aiortc</summary>
<pre>                            if self._debug:
                                logger.warning(msg)
                            continue
&gt;                       raise OSError(err.errno, msg) from None
E                       PermissionError: [Errno 1] error while attempting to bind on address (&#x27;127.0.0.1&#x27;, 1234): [errno 1] operation not permitted

/nix/store/y8z1v4rppi8z436xr3jlpyv8izba0yp5-python3-3.12.12/lib/python3.12/asyncio/base_events.py:1584: PermissionError
=============================== warnings summary ===============================
tests/test_rtcdtlstransport.py: 20 warnings
tests/test_rtcrtpreceiver.py: 24 warnings
tests/test_rtcrtpsender.py: 34 warnings
tests/test_rtcsctptransport.py: 86 warnings
  /nix/store/fwngqngi0zvvqf2n0nsyxz3kiz2z7h72-python3.12-aiortc-1.14.0/lib/python3.12/site-packages/aiortc/rtcdtlstransport.py:206: DeprecationWarning: Attempting to mutate a Context after a Connection was created. In the future, this will raise an exception
    ctx.set_tlsext_use_srtp(b&quot;:&quot;.join(x.openssl_profile for x in srtp_profiles))

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/test_contrib_signaling.py::SignalingTest::test_tcp_socket - PermissionError: [Errno 1] error while attempting to bind on address (&#x27;127....
FAILED tests/test_contrib_signaling.py::SignalingTest::test_tcp_socket_abrupt_disconnect - PermissionError: [Errno 1] error while attempting to bind on address (&#x27;127....
=========== 2 failed, 370 passed, 164 warnings in 140.07s (0:02:20) ============</pre>
</details>
<details>
<summary>python313Packages.aiortc</summary>
<pre>                            if self._debug:
                                logger.warning(msg)
                            continue
&gt;                       raise OSError(err.errno, msg) from None
E                       PermissionError: [Errno 1] error while attempting to bind on address (&#x27;127.0.0.1&#x27;, 1234): [errno 1] operation not permitted

/nix/store/a8qf8fapk2k3b5v26fic1h5h6h88vjhq-python3-3.13.9/lib/python3.13/asyncio/base_events.py:1630: PermissionError
=============================== warnings summary ===============================
tests/test_rtcdtlstransport.py: 20 warnings
tests/test_rtcrtpreceiver.py: 24 warnings
tests/test_rtcrtpsender.py: 34 warnings
tests/test_rtcsctptransport.py: 86 warnings
  /nix/store/7xmqbkqkg5pfh09w8p424narkpbayyq3-python3.13-aiortc-1.14.0/lib/python3.13/site-packages/aiortc/rtcdtlstransport.py:206: DeprecationWarning: Attempting to mutate a Context after a Connection was created. In the future, this will raise an exception
    ctx.set_tlsext_use_srtp(b&quot;:&quot;.join(x.openssl_profile for x in srtp_profiles))

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/test_contrib_signaling.py::SignalingTest::test_tcp_socket - PermissionError: [Errno 1] error while attempting to bind on address (&#x27;127....
FAILED tests/test_contrib_signaling.py::SignalingTest::test_tcp_socket_abrupt_disconnect - PermissionError: [Errno 1] error while attempting to bind on address (&#x27;127....
=========== 2 failed, 370 passed, 164 warnings in 132.12s (0:02:12) ============</pre>
</details>
</details>

---
### `aarch64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python312Packages.aiortc</li>
    <li>python312Packages.aiortc.dist</li>
    <li>python313Packages.aiortc</li>
    <li>python313Packages.aiortc.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>python312Packages.aioice</li>
    <li>python312Packages.aioice.dist</li>
    <li>python312Packages.pylibsrtp</li>
    <li>python312Packages.pylibsrtp.dist</li>
    <li>python313Packages.aioice</li>
    <li>python313Packages.aioice.dist</li>
    <li>python313Packages.pylibsrtp</li>
    <li>python313Packages.pylibsrtp.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>python312Packages.aiortc</summary>
<pre>                            if self._debug:
                                logger.warning(msg)
                            continue
&gt;                       raise OSError(err.errno, msg) from None
E                       PermissionError: [Errno 1] error while attempting to bind on address (&#x27;127.0.0.1&#x27;, 1234): [errno 1] operation not permitted

/nix/store/9m8pfvyz8v5nxck085b1mix0rcpnrm4s-python3-3.12.12/lib/python3.12/asyncio/base_events.py:1584: PermissionError
=============================== warnings summary ===============================
tests/test_rtcdtlstransport.py: 20 warnings
tests/test_rtcrtpreceiver.py: 24 warnings
tests/test_rtcrtpsender.py: 34 warnings
tests/test_rtcsctptransport.py: 86 warnings
  /nix/store/28f7nq31fbpp8r1xiwcn3g6la2z02hn5-python3.12-aiortc-1.14.0/lib/python3.12/site-packages/aiortc/rtcdtlstransport.py:206: DeprecationWarning: Attempting to mutate a Context after a Connection was created. In the future, this will raise an exception
    ctx.set_tlsext_use_srtp(b&quot;:&quot;.join(x.openssl_profile for x in srtp_profiles))

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/test_contrib_signaling.py::SignalingTest::test_tcp_socket - PermissionError: [Errno 1] error while attempting to bind on address (&#x27;127....
FAILED tests/test_contrib_signaling.py::SignalingTest::test_tcp_socket_abrupt_disconnect - PermissionError: [Errno 1] error while attempting to bind on address (&#x27;127....
============ 2 failed, 370 passed, 164 warnings in 98.02s (0:01:38) ============</pre>
</details>
<details>
<summary>python313Packages.aiortc</summary>
<pre>                            if self._debug:
                                logger.warning(msg)
                            continue
&gt;                       raise OSError(err.errno, msg) from None
E                       PermissionError: [Errno 1] error while attempting to bind on address (&#x27;127.0.0.1&#x27;, 1234): [errno 1] operation not permitted

/nix/store/hv7qldswrqvg1sjys1lmgjg875hkffng-python3-3.13.9/lib/python3.13/asyncio/base_events.py:1630: PermissionError
=============================== warnings summary ===============================
tests/test_rtcdtlstransport.py: 20 warnings
tests/test_rtcrtpreceiver.py: 24 warnings
tests/test_rtcrtpsender.py: 34 warnings
tests/test_rtcsctptransport.py: 86 warnings
  /nix/store/hhmpl355y87fxfp2qzy812xhhji71dyd-python3.13-aiortc-1.14.0/lib/python3.13/site-packages/aiortc/rtcdtlstransport.py:206: DeprecationWarning: Attempting to mutate a Context after a Connection was created. In the future, this will raise an exception
    ctx.set_tlsext_use_srtp(b&quot;:&quot;.join(x.openssl_profile for x in srtp_profiles))

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/test_contrib_signaling.py::SignalingTest::test_tcp_socket - PermissionError: [Errno 1] error while attempting to bind on address (&#x27;127....
FAILED tests/test_contrib_signaling.py::SignalingTest::test_tcp_socket_abrupt_disconnect - PermissionError: [Errno 1] error while attempting to bind on address (&#x27;127....
============ 2 failed, 370 passed, 164 warnings in 93.73s (0:01:33) ============</pre>
</details>
</details>

--- a/maintainers/maintainer-list.nix
+++ b/maintainers/maintainer-list.nix
@@ -9418,6 +9418,15 @@
     githubId = 10353047;
     name = "Tobias Happ";
   };
+  gesperon = {
+    email = "gesperon@mac.com";
+    github = "gesperon";
+    githubId = 117491118;
+    name = "Gabor Esperon";
+    keys = [
+      { fingerprint = "8102 3AF7 84FC 4DA7 B9D7  ECFF A0AF 4AD2 D44C C6FE"; }
+    ];
+  };
   getchoo = {
     name = "Seth Flynn";
     email = "getchoo@tuta.io";

--- a/pkgs/development/python-modules/aioice/default.nix
+++ b/pkgs/development/python-modules/aioice/default.nix
@@ -1,0 +1,51 @@
+{
+  lib,
+  buildPythonPackage,
+  setuptools,
+  dnspython,
+  ifaddr,
+  fetchFromGitHub,
+  pytestCheckHook,
+}:
+
+buildPythonPackage rec {
+  pname = "aioice";
+  version = "0.10.1";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "aiortc";
+    repo = "aioice";
+    tag = version;
+    hash = "sha256-KFYPzGPm+d1QrFAW9OhTDxroV/MnFusmfy5qcYCfDiM=";
+  };
+
+  build-system = [
+    setuptools
+  ];
+
+  dependencies = [
+    dnspython
+    ifaddr
+  ];
+
+  nativeCheckInputs = [ pytestCheckHook ];
+  disabledTestPaths = [
+    # Network tests failing
+    "tests/test_ice.py"
+    "tests/test_mdns.py"
+    "tests/test_turn.py"
+    "tests/test_ice_trickle.py"
+  ];
+  doCheck = true;
+  pythonImportsCheck = [
+    "aioice"
+  ];
+
+  meta = {
+    description = "Asyncio-based Interactive Connectivity Establishment (RFC 5245)";
+    homepage = "https://github.com/aiortc/aioice";
+    license = lib.licenses.bsd3;
+    maintainers = with lib.maintainers; [ gesperon ];
+  };
+}

--- a/pkgs/development/python-modules/aiortc/default.nix
+++ b/pkgs/development/python-modules/aiortc/default.nix
@@ -1,0 +1,74 @@
+{
+  lib,
+  buildPythonPackage,
+  setuptools,
+  aioice,
+  av,
+  cffi,
+  cryptography,
+  google-crc32c,
+  pyee,
+  pylibsrtp,
+  pyopenssl,
+  libopus,
+  libvpx,
+  ifaddr,
+  dnspython,
+  fetchFromGitHub,
+  pytestCheckHook,
+  numpy,
+}:
+
+buildPythonPackage rec {
+  pname = "aiortc";
+  version = "1.14.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "aiortc";
+    repo = "aiortc";
+    tag = version;
+    hash = "sha256-ZgxSaiKkJrA5XvUT1zq8kwqB8mOvn46vLWXHyJSsHbM=";
+  };
+
+  build-system = [
+    setuptools
+  ];
+
+  dependencies = [
+    aioice
+    av
+    cffi
+    cryptography
+    google-crc32c
+    pyee
+    pylibsrtp
+    pyopenssl
+    libopus
+    libvpx
+    ifaddr
+    dnspython
+  ];
+
+  nativeCheckInputs = [
+    numpy
+    pytestCheckHook
+  ];
+  disabledTestPaths = [
+    "tests/test_ortc.py" # hangs on: aiortc.rtcicetransport:rtcicetransport.py:365 RTCIceTransport(controlled) - new -> checking
+    "tests/test_rtcicetransport.py" # hangs on: aiortc.rtcicetransport:rtcicetransport.py:365 RTCIceTransport(controlled) - new -> checking
+    "tests/test_rtcpeerconnection.py" # fails
+    "tests/test_contrib_signaling.py" # fails on darwin
+  ];
+  doCheck = true;
+  pythonImportsCheck = [
+    "aiortc"
+  ];
+
+  meta = {
+    description = "WebRTC and ORTC implementation for Python using asyncio";
+    homepage = "https://github.com/aiortc/aiortc";
+    license = lib.licenses.bsd3;
+    maintainers = with lib.maintainers; [ gesperon ];
+  };
+}

--- a/pkgs/development/python-modules/pylibsrtp/default.nix
+++ b/pkgs/development/python-modules/pylibsrtp/default.nix
@@ -1,0 +1,46 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  setuptools,
+  cffi,
+  srtp,
+  openssl,
+  pytestCheckHook,
+}:
+
+buildPythonPackage rec {
+  pname = "pylibsrtp";
+  version = "1.0.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "aiortc";
+    repo = "pylibsrtp";
+    tag = version;
+    hash = "sha256-Q8EyGAJKkq14sqSEMWLB8arKvj/wuALK/XwOZ27F1nQ=";
+  };
+
+  build-system = [
+    setuptools
+  ];
+
+  dependencies = [
+    cffi
+    srtp
+    openssl
+  ];
+
+  nativeCheckInputs = [ pytestCheckHook ];
+  doCheck = true;
+  pythonImportsCheck = [
+    "pylibsrtp"
+  ];
+
+  meta = {
+    description = "Python bindings for libsrtp";
+    homepage = "https://github.com/aiortc/pylibsrtp";
+    license = lib.licenses.bsd3;
+    maintainers = with lib.maintainers; [ gesperon ];
+  };
+}

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -13594,6 +13594,8 @@ self: super: with self; {
 
   pylibrespot-java = callPackage ../development/python-modules/pylibrespot-java { };
 
+  pylibsrtp = callPackage ../development/python-modules/pylibsrtp { };
+
   pylink-square = callPackage ../development/python-modules/pylink-square { };
 
   pylint = callPackage ../development/python-modules/pylint { };

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -348,6 +348,8 @@ self: super: with self; {
 
   aiohwenergy = callPackage ../development/python-modules/aiohwenergy { };
 
+  aioice = callPackage ../development/python-modules/aioice { };
+
   aioimaplib = callPackage ../development/python-modules/aioimaplib { };
 
   aioimmich = callPackage ../development/python-modules/aioimmich { };

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -460,6 +460,8 @@ self: super: with self; {
 
   aiorpcx = callPackage ../development/python-modules/aiorpcx { };
 
+  aiortc = callPackage ../development/python-modules/aiortc { };
+
   aiortm = callPackage ../development/python-modules/aiortm { };
 
   aiortsp = callPackage ../development/python-modules/aiortsp { };

